File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredStringPropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KRequiredStringPropertyName implements KRequiredPropertyName<String>, KSortablePropertyName<String> {

    private final String name;

    public KRequiredStringPropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredStringPropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KRequiredStringPropertyName(val name: String) : KRequiredPropertyName<String>, KSortablePropertyName<String> {

    override fun name(): String = name

}
